### PR TITLE
Add Cloudant Query markdown document

### PR DESCRIPTION
**Pull request by alfinkel**
This PR adds documentation for the new Cloudant Query functionality and redirects the link in the main sync-android readme.md from the old index-query.md to the new query.md.  Also, added empty lines at the end of all Query Java classes that did not have them before.  This was noted in a previous comment as a need to bring the code more in line with our coding standards.

### CODE REVIEW

File: doc/query.md [hunk 1]
```diff
@@ -0,0 +1,545 @@
+# Cloudant Query - Android
+
+Cloudant Query - Android is a Java implementation of [Cloudant Query][1] for Android mobile devices. It is included as a sub-package of [sync-android][2].
```
Comment (mikerhodes) [inline doc/query.md]: It's in the sync-core jar, so I'd use that here rather than the repo name, which is ambiguous.

File: doc/query.md [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+
+Cloudant Query is inspired by MongoDB's query implementation, so users of MongoDB should feel at home using Cloudant Query in their mobile applications.
+
+The aim is that the query you use on our cloud-based database works for your mobile application.
+
+**Note**: this is a very early version, and it probably won't work as you expect yet! In
+particular, there's very little information to help debugging when errors happen, leaving you
+in the debugger. There is also little error checking, so things might explode without warning.
```
Comment (mikerhodes) [inline doc/query.md]: I don't think this is true any more. Remove from here, and we can also remove from the objc version.

File: doc/query.md [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+
+#### Sorting
+
+Provide a sort document to the extended version of the `find` method to sort the results of a query. 
+
+The sort document is a list of fields to sort by. Each field is represented by a map specifying the name of the field to sort by and the direction to sort.
+
+The sort document must use fields from a single index.
+
+As yet, you can't leave out the sort direction. The sort direction can be `asc` (ascending) or `desc` (descending).
```
Comment (mikerhodes) [inline doc/query.md]: Is this still true, "As yet, you can't leave out the sort direction"?

File: doc/query.md [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+    "pet": { "species": "cat" }
+}
+```
+
+To project the `name` and `age` fields of the above document:
+
+```java
+List<String> fields = new ArrayList<String>();
+fields.add("name");
+fields.add("age");
```
Comment (mikerhodes) [inline doc/query.md]: Can we use the `List<String> fields = Arrays.asList("name", "age")` syntax here?

File: doc/query.md [hunk 5]
```diff
@@ -0,0 +1,545 @@
...
+The document _would_ be indexed in both of these indexes: each index only contains one of
+the array fields.
+
+Also see "Unsupported features", below.
+
+
+### Errors
+
+Error reporting is terrible right now. The only indication something went wrong is a `null` return value from the `find` methods or `ensureIndexed(List<Object> fieldNames, String indexName)`. We're working on
+adding logging.
```
Comment (mikerhodes) [inline doc/query.md]: You can probably say "We log the errors, but don't throw any exceptions" now.

File: doc/query.md [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+- Querying to match a specific array element using dotted notation, `{ field.0: 1 }`.
+- Querying using `$all`.
+- Querying using `$elemMatch`.
+
+
+## Performance
+
+### Indexing
+
+N/A
```
Comment (mikerhodes) [inline doc/query.md]: I'd spell out "Not carried out yet" rather than "Not Applicable".

File: doc/query.md [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+<em>boolean</em> := <em>Boolean</em>
+
+<em>type</em> := <em>Class</em>
+</pre>
+
+
+
+## Requirements
+
+This package as well as all of its dependencies co-exist in the [Cloudant Sync - Android Datastore][2].  Installing that package by default installs this sub-package.
```
Comment (mikerhodes) [inline doc/query.md]: Probably can just remove this section.

File: doc/query.md [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+
+## License
+
+Cloudant Query - Android is available under the Apache V2 license. See the LICENSE file for more info.
```
Comment (mikerhodes) [inline doc/query.md]: This isn't needed either as it's covered by the package-as-a-whole license.

File: doc/query.md (hunk 3 above)
Comment (alfinkel) [inline doc/query.md]: Yes this is still true (both Android and iOS).

File: doc/query.md (hunk 1 above)
Comment (alfinkel) [inline doc/query.md]: Updated in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3

File: doc/query.md (hunk 2 above)
Comment (alfinkel) [inline doc/query.md]: Removed in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3.  Will make a note to remove from objc version.

File: doc/query.md (hunk 4 above)
Comment (alfinkel) [inline doc/query.md]: Done in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3

File: doc/query.md (hunk 5 above)
Comment (alfinkel) [inline doc/query.md]: Updated in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3

File: doc/query.md (hunk 6 above)
Comment (alfinkel) [inline doc/query.md]: I was using N/A as Not Available.  Updated to "Not carried out yet" to remove any ambiguity https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3

File: doc/query.md (hunk 7 above)
Comment (alfinkel) [inline doc/query.md]: Done in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3

File: doc/query.md (hunk 8 above)
Comment (alfinkel) [inline doc/query.md]: Removed in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3

**Pull request merged**